src/python/pants/backend/scala/lint/scalafix/rules.py

```diff
@@ -321,7 +323,14 @@ async def _run_scalafix_process(
                         else ()
                     ),
                     *(("--check",) if request.check_only else ()),
-                    *((f"--scalac-options={arg}" for arg in scalac.args) if scalac.args else ()),
+                    *(
+                        (
+                            f"--scalac-options={arg}"
+                            for arg in scalac.parsed_args_for_resolve(partition_info.resolve_name)
+                        )
+                        if scalac.parsed_args_for_resolve(partition_info.resolve_name)
```

Comment:
Probably better to capture `scalac.parsed_args_for_resolve(partition_info.resolve_name)` into a variable rather than invoke twice?